src/app/content/utils/urlUtils.ts

```diff
@@ -74,15 +29,15 @@ export const getUrlParamForPageId = (book: Pick<Book, 'id' | 'tree' | 'title'>,
   if (!treeSection) {
     throw new Error(`BUG: could not find page "${pageId}" in ${book.title}`);
   }
-  const result = getUrlParamForPageTitle(treeSection);
+  const result = assertDefined(treeSection.slug, 'Slug needs to be defined');
   getUrlParamForPageIdCache.set(cacheKey, result);
 
   return result;
 };
 
 export const getPageIdFromUrlParam = (book: Book, pageParam: string): string | undefined => {
   for (const section of flattenArchiveTree(book.tree)) {
-    const sectionParam = getUrlParamForPageTitle(section);
+    const sectionParam = assertDefined(section.slug, 'Slug needs to be defined');
```

Comment:
can we put the book/page information in these error messages?

Reply:
yes, updated.